pkg/query-service/app/integrations/messagingQueues/kafka/consumerLag.md

```diff
@@ -0,0 +1,197 @@
+## Consumer Lag feature break down
+
+### 1) Consumer Lag Graph
+
+
+---
+
+### 2) Consumer Group Details
+
+API endpoint:
+
+```
+POST /api/v1/messaging-queues/kafka/consumer-lag/consumer-details
+```
+
+```json
+{
+	"start": 1720685296000000000,
+	"end": 1721290096000000000,
+	"variables": {
+        "partition": "0",
+		"topic": "topic1",
+        "consumer_group": "cg1"
+	}
+}
+```
+
+response in query range format `series`
+```json
+{
+    "status": "success",
+    "data": {
+        "resultType": "",
+        "result": [
+            {
+                "table": {
+                    "columns": [
+                        {
+                            "name": "service_name",
+                            "queryName": "",
+                            "isValueColumn": false
+                        },
+                        {
+                            "name": "p99",
+                            "queryName": "",
+                            "isValueColumn": false
```

Comment:
@vikrantgupta25 I don't recall what is `isValueColumn` used for. What would break if we send a value column with `isValueColumn` false?

Reply:
@srikanthccv this was used to apply legends to the queries. let's say i check the latency of the services and `group_by` service name then we get 2 columns service_name and the value column A. and if we have some custom legend for the query then we check if it is a value column and then apply the same based on query name passed as well

Reply:
Should I mark this resolved? or do I need to sanitise the existing query_range `tabular` response for messaging queues?

Reply:
This can be resolved because there is no user input needed.